library/types/src/modules/Hostname.rb

```diff
@@ -159,7 +160,7 @@ def CurrentFQ
         SCR.Execute(path(".target.bash_output"), "hostname --fqdn")
       )
       if hostname_data.nil? || Ops.get_integer(hostname_data, "exit", -1) != 0
-        fqhostname = if SCR.Read(path(".target.stat"), "/etc/hostname").empty?
+        fqhostname = if FileUtils.Exists("/etc/hostname")
```

Comment:
I think it is problem during upgrade, where you want to read hostname from target machine